src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.cs

```diff
@@ -207,7 +206,9 @@ public void EndInit_ResumesPausedEnableRaisingEvents(bool setBeforeBeginInit)
                 if (!setBeforeBeginInit)
                     watcher.EnableRaisingEvents = true;
                 watcher.EndInit();
-                ExpectEvent(watcher, WatcherChangeTypes.Created | WatcherChangeTypes.Deleted, () => new TempFile(Path.Combine(testDirectory.Path, GetTestFileName())).Dispose(), null);
+                ExpectEvent(
+                    watcher, WatcherChangeTypes.Created | WatcherChangeTypes.Deleted, () => new TempFile(Path.Combine(testDirectory.Path, GetTestFileName())).Dispose(),
+                    null, (string)null, 10, 1000); // https://github.com/dotnet/runtime/issues/24181 for the reason for different constants
```

Comment:
previously we would have up to three retries:

https://github.com/dotnet/runtime/blob/a7d1332df4f33ca4f9d4d7e5e979ecdb596b8f0a/src/libraries/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs#L29

with a `500 ms` timeout:

https://github.com/dotnet/runtime/blob/a7d1332df4f33ca4f9d4d7e5e979ecdb596b8f0a/src/libraries/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs#L24

so in theory `1.5 s` would be the max time it takes to execute this test. With the proposed change it would be up to `10 s (10 * 1_000 ms)`. 

How frequently can we hit the most pessimistic scenario (10s)? Is this a common problem, or a very rare situation? Have we done some debugging to identify the source of the actual problem?

Reply:
Maybe we can consider a change like this (maybe not so drastic as 10*1000, but smaller) if we add the test method to outerloop. @lukas-lansky